### ISSUE TITLE
Short Circuit shop bug

### ISSUE THREAD

**Issue by TerryCavanagh** [TerryCavanagh/diceydungeons.com#464]
![buy_short_circuit](https://user-images.githubusercontent.com/2874150/44944638-3f355f80-add2-11e8-8054-cc0c175c698e.png)

![2018-08-31_21-07-06](https://user-images.githubusercontent.com/2874150/44944701-6a6c7e80-add3-11e8-9534-3434ba05cfc6.gif)

Buying short circuit from the shop as a robot fails, and causes a visual glitch.

```
I was in a store, and I bought Short Circuit. It made the "Buy" button appear 4 times, and I didn't receive the item, but it took my gold

...

where the "buy" button was, there was suddenly a new copy of the buy button to the left of it, and two more copies to the right of it. It got drawn 3 extra times.

basically, it didn't take me to the "Equip, Backpack, Ignore" screen after I bought the item
```

This also happens with the Witch apple shop:

![buy_apple_display](https://user-images.githubusercontent.com/2874150/44944651-63913c00-add2-11e8-9e5a-009b632ea8b4.png)

```
the apple on the right can't be bought at this point (there were three available)
```

**Comment by TerryCavanagh** [comment]
Fixed.
